website/client/components/achievements/wonChallenge.vue

```diff
@@ -58,6 +58,12 @@ export default {
     };
   },
   methods: {
+    async syncUser () {
+      // @TODO: There are times when the user is not updated when this modal is shown
+      // It would be nice to not need to fetch to full user, maybe just the achievements?
+      // See https://github.com/HabitRPG/habitica/issues/7716
+      await this.$store.dispatch('user:fetch', {forceLoad: true});
```

Comment:
should something be executed after the dispatch? if not you can remove the `async`/`await` and it'll still work ^^

Reply:
Only if I return the promise and await this function.